configure.ac

```diff
@@ -729,28 +729,26 @@ AC_ARG_WITH(
       iOS)
         # Always apply the compliance patch on iOS; we can use the macOS patch
         APP_STORE_COMPLIANCE_PATCH="Mac/Resources/app-store-compliance.patch"
-        APP_STORE_COMPLIANCE_PATCH_TARGET="build/app-store-compliant"
         APP_STORE_COMPLIANCE_PATCH_FLAGS=
+        INSTALLTARGETS="$INSTALLTARGETS app-store-compliant"
         AC_MSG_RESULT([applying default app store compliance patch])
         ;;
       Darwin)
         # Always *check* the compliance patch on macOS
         APP_STORE_COMPLIANCE_PATCH="Mac/Resources/app-store-compliance.patch"
-        APP_STORE_COMPLIANCE_PATCH_TARGET="build/app-store-compliant"
         APP_STORE_COMPLIANCE_PATCH_FLAGS="--dry-run"
```

Comment:
While reviewing the Makefile changes, I was reminded of the `configure` `--disable-test-modules` option. Specifying that causes the Makefile to be generated to not install test directories and files, something which seems like an option that developers of stand-alone apps with an embedded Python might be likely to use.  Unfortunately, that causes problems with the current approach of having a single unified default patch file for iOS and macOS builds (i.e. `app-store-compliance.patch`) since it includes patches for both a non-test and a test file. If a file to be patched is not present, `patch` fails (somewhat more gracefully if the --batch option is included):

```
No file to patch.  Skipping...
1 out of 1 hunks ignored--saving rejects to 'test/test_urlparse.py.rej'
Can't create '/var/folders/88/pzbbx0t17b58mtgk9hp9v4jh0000gn/T/patchrkhI75QBojY', output is in '/var/folders/88/pzbbx0t17b58mtgk9hp9v4jh0000gn/T/patchrkhI75QBojY': No such file or directory
patching file 'urllib/parse.py'
make: *** [app-store-compliant] Error 1
```

So the net effect for this patch is kinda OK, since urlib/parse.py was successfully patched and the `test/test_urlparse.py.rej` cannot be created since there is no `test` directory. Of course, the user can always specify a different (i.e. edited) patch file that doesn't include the `test` file patch.  As a somewhat ugly hack, we could ignore `patch` errors and continue on.

It appears that `patch --dry-run` also fails if the file to be patched does not exist, which means that `make install` will fail for macOS builds if `--disable-test-modules` is included on `configure` and the default patch file is not overridden with `--with-app-store-compliance=...`.  So even with other changes, it might be best to not try to run `patch --dry-run` by default on macOS builds.

Reply:
I didn't know about `--disable-test-modules` - good to know. 

Ignoring patch errors seems like a bad idea to me - we *want* the patch to be noisy if it can't be applied, so that we know the patch needs to be updated.

Maybe we could split the difference - *always* do a `--dry-run` on the app sources directory (so we can verify that the patch is up to date), but then only *actually* apply the patch if it is required, ignoring patch errors and missing files. Does that make sense?

Reply:
I suppose that could work assuming we can trust `patch` to not try to modify anything and to have the required options. It's a bit worrying that there are various flavors of `patch` out there including a POSIX standard. Also, the vast majority of builders of Python are not going to be using the patch.